rest/rest-server/src/main/java/org/apache/brooklyn/rest/RestApiSetup.java

```diff
@@ -54,6 +63,21 @@ public static void installRest(ServletContextHandler context, Object... provider
         context.addServlet(servletHolder, "/v1/*");
     }
 
+    private static void injectManagementContext(ManagementContext managementContext, Object o) {
+        if (managementContext != null && o instanceof ManagementContextInjectable) {
+            ((ManagementContextInjectable) o).setManagementContext(managementContext);
+        }
+    }
+
+    private static ManagementContext extractManagementContext(Object... providers) {
+        for (Object o : providers) {
+            if (o instanceof ManagementContextProvider) {
+                return ((ManagementContextProvider) o).getContext(ManagementContext.class);
+            }
+        }
+        return null;
+    }
+
```

Comment:
I suggest we apply this PR after #41, so I can look at the failing test and see how to make it work with the `ManagementContextProvider`